backend/app/Chat/User/UserLock.php

```
@@ -0,0 +1,201 @@
+                    $lockAcquired = true;
+                    break;
+                }
+
+                if ($i < $maxRetries - 1) {
+                    usleep($retryDelay);
+                    $retryDelay *= 2; // Exponential backoff
+                }
+            }
+
+            if (!$lockAcquired) {
+                throw new \Exception('Failed to acquire user lock after ' . $maxRetries . ' attempts');
```

PHPMD (2.15.0)

Missing class import via use statement (line '187', column '27'). (Clean Code Rules)

(MissingImport)